kolibri/core/auth/api.py

```diff
@@ -482,39 +515,66 @@ def perform_bulk_destroy(self, objects):
             raise PermissionDenied("Super user cannot delete self")
         objects.update(date_deleted=now())
 
-    def consolidate(self, items, queryset):
-        output = []
-        items = sorted(items, key=lambda x: x["id"])
-        ordering_param = self.request.query_params.get("ordering", self.order_by_field)
-        reverse = False
-        for key, group in groupby(items, lambda x: x["id"]):
-            roles = []
-            for item in group:
-                role = {
-                    "collection": item.pop("roles__collection"),
-                    "kind": item.pop("roles__kind"),
-                    "id": item.pop("roles__id"),
-                }
-                if role["collection"]:
-                    # Our values call will return null for users with no assigned roles
-                    # So filter them here.
-                    roles.append(role)
-            item["roles"] = roles
-            output.append(item)
-            if ordering_param.startswith("-"):
-                ordering_param = ordering_param[1:]
-                reverse = True
-
-        output = sorted(output, key=lambda x: x[ordering_param], reverse=reverse)
-        return output
-
     def perform_update(self, serializer):
         instance = serializer.save()
         # if the user is updating their own password, ensure they don't get logged out
         if self.request.user == instance:
             update_session_auth_hash(self.request, instance)
 
 
+class DeletedFacilityUserViewSet(
+    FacilityUserConsolidateMixin,
+    ReadOnlyValuesViewset,
+    DestroyModelMixin,
+    BulkDeleteMixin,
+):
+    """Viewset for managing soft-deleted FacilityUsers."""
+
+    permission_classes = (KolibriAuthPermissions,)
+    pagination_class = OptionalPageNumberPagination
+    filter_backends = (
+        KolibriAuthPermissionsFilter,
+        DjangoFilterBackend,
+        filters.SearchFilter,
+        ValuesViewsetOrderingFilter,
+    )
+
+    order_by_field = "-date_deleted"
+    queryset = FacilityUser.soft_deleted_objects.all().order_by(order_by_field)
+    serializer_class = FacilityUserSerializer
+    filterset_class = FacilityUserFilter
+
+    search_fields = FacilityUserViewSet.search_fields
+    values = FacilityUserViewSet.values + ("date_deleted",)
+    ordering_fields = FacilityUserViewSet.ordering_fields + ("date_deleted",)
+    field_map = FacilityUserViewSet.field_map
+
+    def allow_bulk_restore(self):
+        """
+        Hook to ensure that the bulk restore should be allowed.
+        By default, these restrictions are the same as for bulk deletion,
+        """
+        return self.allow_bulk_destroy(self.get_queryset())
+
+    @decorators.action(detail=False, methods=["post"])
+    def restore(self, request):
+        """
+        Restore soft-deleted FacilityUsers.
+        """
+        if not self.allow_bulk_restore():
```

Comment:
If we modify allow_bulk_destroy as per my comments below, this can just become:
```
if not self.allow_bulk_destroy():
```

Reply:
Done!